docs/openapi.yaml

```diff
@@ -0,0 +1,310 @@
+      tags:
+      - System
+  /_version:
+    get:
+      operationId: get_version__version_get
+      responses:
+        '200':
+          content:
+            application/json:
+              schema: {}
+          description: Successful Response
+      summary: Get Version
+      tags:
+      - System
+  /log/login:
+    post:
+      description: 'Create a new `login` audit log.
+
+
+        This endpoint does not include any authorization checks, but it is not
+
+        exposed and is only meant for internal use.
+
+
+        If the timestamp is omitted from the request body, the current date and
+
+        time will be used.
+
+
+        The response is returned _before_ inserting the new audit log in the
+
+        database, so that POSTing audit logs does not impact the performance of
+
+        the caller and audit-service failures are not visible to users.'
+      operationId: create_login_log_log_login_post
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateLoginLogInput'
+        required: true
+      responses:
+        '201':
+          content:
+            application/json:
+              schema: {}
+          description: Successful Response
+        '422':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPValidationError'
+          description: Validation Error
+      security:
+      - HTTPBearer: []
+      summary: Create Login Log
+      tags:
+      - Maintain
+  /log/presigned_url:
+    post:
+      description: 'Create a new `presigned_url` audit log.
+
+
+        This endpoint does not include any authorization checks, but it is not
+
+        exposed and is only meant for internal use.
+
+
+        If the timestamp is omitted from the request body, the current date and
+
+        time will be used.
+
+
+        The response is returned _before_ inserting the new audit log in the
+
+        database, so that POSTing audit logs does not impact the performance of
+
+        the caller and audit-service failures are not visible to users.'
+      operationId: create_presigned_url_log_log_presigned_url_post
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreatePresignedUrlLogInput'
+        required: true
+      responses:
+        '201':
+          content:
+            application/json:
+              schema: {}
+          description: Successful Response
+        '422':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPValidationError'
+          description: Validation Error
+      security:
+      - HTTPBearer: []
+      summary: Create Presigned Url Log
+      tags:
+      - Maintain
+  /log/{category}:
+    get:
+      description: "Queries the logs the current user has access to see. Returned\
```

Comment:
I made a free test account on Swagger and it likes like we may be able to get more natural formatting. This seems valid (and it auto-generated it for me too!) ✨ 

```
      description: >-
        Queries the logs the current user has access to see. Returned data:

            {
                "nextTimeStamp": <timestamp or null>,
                "data": [<entry>, <entry>, ...] OR int if using `count` param,
            }

        This endpoint only returns up to a configured maximum number of entries

        at a time. If there are more entries to query, it returns a non-null

        "nextTimeStamp" which can be used to get the next page.


        The returned entries are ordered by increasing timestamp (least recent
        to

        most recent), so that new entries are at the end and there is no risk of

        skipping entries when getting the next page.


        Filters can be added as query strings. Accepted filters include all
        fields

        for the queried category, as well as the following special filters:

        - "groupby" to get counts

        - "count" to get the number of rows instead of a list

        - "start" to specify a starting timestamp (inclusive). Default: none

        - "stop" to specify an end timestamp (exclusive). Default: none


        If queries are time-boxed (depends on the configuration),

        ("stop" - "start") must be lower than the configured maximum.


        Without filters, this endpoint will return all data within the time-box.

        Add filters as query strings like this:

            GET /log/presigned_url?a=1&b=2

        This will match all records that have values containing all of:

            {"a": 1, "b": 2}

        Providing the same key with more than one value filters records whose

        value of the given key matches any of the given values. But values of

        different keys must all match. For example:

            GET /log/presigned_url?a=1&a2&b=3

        Matches these:

            {"a": 1, "b": 3}
            {"a": 2, "b": 3}

        But won't match these:

            {"a": 1, "b": 10}
            {"a": 10, "b": 3}

        `groupby` example:

            GET /log/presigned_url?a=1&groupby=b&groupby=c

            {"b": 1, "c": 2, "count": 5}
            {"b": 1, "c": 3, "count": 8}

        `count` example:

            GET /log/presigned_url?a=1&groupby=b&groupby=c&count

            Returns: 2 (see previous example returning 2 rows)
```

Reply:
the swagger doc is generated automatically from the docstring: https://github.com/uc-cdis/audit-service/blob/aa54b60890578e24e2d58531775692a4072cd0f7/src/audit/routes/query.py#L31

Reply:
✔️

Reply:
but yeah this file doesn't look great 😅